src/main/java/core/basesyntax/FigureSupplier.java

```diff
@@ -0,0 +1,37 @@
+package core.basesyntax;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Random;
+
+public class FigureSupplier {
+    private final Random random = new Random();
+
+    public Figure getRandomFigure() {
+        ColorSupplier supplier = new ColorSupplier();
+        Figure f1 = new Circle(getRandomProperty(), supplier.getRandomColor());
+        Figure f2 = new Square(getRandomProperty(), getRandomProperty(),
+                supplier.getRandomColor());
+        Figure f3 = new RightTriangle(getRandomProperty(), getRandomProperty(),
+                getRandomProperty(), supplier.getRandomColor());
+        Figure f4 = new Rectangle(getRandomProperty(), getRandomProperty(),
+                supplier.getRandomColor());
+        Figure f5 = new IsoscelesTrapezoid(getRandomProperty(), getRandomProperty(),
+                getRandomProperty(), supplier.getRandomColor());
+        List<Figure> figures = new ArrayList<>();
```

Comment:
You create 5 figures but need only 1.
1.  Make your choice
2. Create the needed Figure

Reply:
`return figures.get(random.nextInt(figures.size()));`

I created 5 different figures with random properties, and added to list, and then take one from list with random index

Reply:
dont create redundant objects, randomize number, and then create just 1 object

Reply:
you are creating too many objects